vortex-array/src/arrays/primitive2/mod.rs

```diff
@@ -0,0 +1,102 @@
+#![allow(dead_code)]
+#![allow(unused_variables)]
+
+use arrow_buffer::BooleanBufferBuilder;
+use vortex_buffer::{Buffer, BufferMut};
+use vortex_dtype::{DType, Nullability, PType};
+
+use crate::stats::ArrayStats;
+use crate::validity::Validity;
+
+pub trait PrimitiveType<const NULLABLE: bool>: Sized {
```

Comment:
Could this be the Nullability enum? the bool reads very weird, should be functionaly the same

Reply:
cool idea, I think it should work

Reply:
trying it now

Reply:
If we change it to a pseudo enum with associated consts it can work. It doesn't have to replace the enum I guess? It can be an additional type that can be converted into `Nullability`

Reply:
Oh this has been open only for 3 years